tests/ducktape/producer_strategy.py

```diff
@@ -0,0 +1,299 @@
+"""
+Producer strategies for testing sync and async Kafka producers.
+
+This module contains strategy classes that encapsulate the different producer
+implementations (sync vs async) with consistent interfaces for testing.
+"""
+import time
+import asyncio
+from confluent_kafka import Producer
+
+
+class ProducerStrategy:
+    """Base class for producer strategies"""
+    def __init__(self, bootstrap_servers, logger):
+        self.bootstrap_servers = bootstrap_servers
+        self.logger = logger
+        self.metrics = None
+
+    def create_producer(self):
+        raise NotImplementedError()
+
+    def produce_messages(self, topic_name, test_duration, start_time, message_formatter, delivered_container, failed_container=None):
+        raise NotImplementedError()
+
+
+class SyncProducerStrategy(ProducerStrategy):
+    def create_producer(self, config_overrides=None):
+        config = {
+            'bootstrap.servers': self.bootstrap_servers,
+            # Optimized configuration for low-latency, high-throughput (same as async)
+            'queue.buffering.max.messages': 1000000,  # 1M messages (sufficient)
+            'queue.buffering.max.kbytes': 1048576,    # 1GB (default)
+            'batch.size': 65536,                      # 64KB batches (increased for better efficiency)
+            'batch.num.messages': 50000,              # 50K messages per batch (up from 10K default)
+            'message.max.bytes': 2097152,             # 2MB max message size (up from ~1MB default)
+            'linger.ms': 1,                          # Wait 1ms for batching (low latency)
+            'compression.type': 'lz4',               # Fast compression
+            'acks': 1,                               # Wait for leader only (faster)
+            'retries': 3,                            # Retry failed sends
+            'delivery.timeout.ms': 30000,            # 30s delivery timeout
+            'max.in.flight.requests.per.connection': 5  # Pipeline requests
+        }
+        
+        # Apply any test-specific overrides
+        if config_overrides:
+            config.update(config_overrides)
+            
+        producer = Producer(config)
+        
+        # Log the configuration for validation
+        self.logger.info("=== SYNC PRODUCER CONFIGURATION ===")
+        for key, value in config.items():
+            self.logger.info(f"{key}: {value}")
+        self.logger.info("=" * 40)
+        
+        return producer
+    
+    def produce_messages(self, topic_name, test_duration, start_time, message_formatter, delivered_container, failed_container=None):
```

Comment:
A lot of shared code with Async variant -- we'll want to squash that smaller at some point